sw/device/lib/testing/meson.build

```
@@ -22,10 +22,10 @@ sw_lib_testing_entropy_testutils_lib = declare_dependency(
 )
 
 # Random number generator.
-sw_lib_testing_random = declare_dependency(
+sw_lib_testing_rand_testutils = declare_dependency(
   link_with: static_library(
-    'random_ot',
-    sources: ['random.c'],
+    'sw_lib_rand_testutils',
```

nit: 
```suggestion
    'sw_lib_testing_rand_testutils',
```
to stay consistent with other lib names.

---

Since this is addressing a long list of tasks (#7976), I fixed this in the next PR (see #8039) in the chain.